common/djangoapps/edxmako/paths.py

```diff
@@ -49,15 +54,25 @@ def add_directory(self, directory, prepend=False):
 
     def get_template(self, uri):
         """
-        Overridden method which will hand-off the template lookup to the microsite subsystem
+        Overridden method for locating a template in either the database or the site theme.
+
+        If not found, template lookup will be done in comprehensive theme for current site
+        by prefixing path to theme.
+        e.g if uri is `main.html` then new uri would be something like this `/red-theme/lms/static/main.html`
+
+        If still unable to find a template, it will fallback to the default template directories after stripping off
+        the prefix path to theme.
         """
-        microsite_template = microsite.get_template(uri)
+        template = themed_template(uri)
 
-        return (
-            microsite_template
-            if microsite_template
-            else super(DynamicTemplateLookup, self).get_template(uri)
-        )
+        if not template:
+            try:
```

Comment:
Let's add some more comments here explaining the purpose of the entire conditional, rather than just a single comment for the 'else' condition